kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/sets/SetsIntersection.kt

```diff
@@ -0,0 +1,22 @@
+package io.kotest.matchers.sets
+
+import io.kotest.assertions.print.print
+import io.kotest.matchers.Matcher
+import io.kotest.matchers.MatcherResult
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldNot
+
+infix fun <T> Set<T>.shouldIntersect(ts: Set<T>) = this should intersectWith(ts)
+infix fun <T> Set<T>.shouldNotIntersect(ts: Set<T>) = this shouldNot intersectWith(ts)
+
+fun <T> intersectWith(ts: Set<T>): Matcher<Set<T>> = object : Matcher<Set<T>> {
```

Comment:
🟡 I think we should explain clearer what a Set Intersection means. I know it's mathematically well-defined, but users don't need to remember these concepts when reading through the suggestions from IDE

Reply:
> 🟡 I think we should explain clearer what a Set Intersection means. I know it's mathematically well-defined, but users don't need to remember these concepts when reading through the suggestions from IDE

will do thx

Reply:
> 🟡 I think we should explain clearer what a Set Intersection means. I know it's mathematically well-defined, but users don't need to remember these concepts when reading through the suggestions from IDE

done